src/components/Set/Summary/index.js

```diff
@@ -159,11 +186,29 @@ class SummarySet extends PureComponent /*:: <Props, State> */ {
       this.props.data.metadata.relationships.nodes.length <= MAX_NUMBER_OF_NODES
     )
       this.setState({ showClanViewer: true });
-    this._ref.current.addEventListener('click', this._handleClick);
+    this._ref.current?.addEventListener('click', (evt /*: Event */) =>
+      this._handleClick(evt),
+    );
   }
 
   componentDidUpdate(prevProps) {
-    if (prevProps.data !== this.props.data) this.loaded = false;
+    if (prevProps.label !== this.props.label) this._refreshLabels();
+    if (
+      prevProps.data?.metadata?.accession !==
+      this.props?.data?.metadata?.accession
+    )
+      this.loaded = false;
+    this.repaint();
+  }
+
+  componentWillUnmount() {
+    if (this._ref.current) {
+      this._ref.current?.removeEventListener('click', this._handleClick);
+    }
+    // TODO: Update clanviewer to clean SVG
```

Comment:
If this TODO is still relevant, maybe you could create a GH issue or JIRA ticket so it's not forgotten.